src/storage/src/region/writer.rs

```diff
@@ -561,4 +569,47 @@ impl WriterInner {
 
         Ok(())
     }
+
+    fn build_flush_callback<S: LogStore>(
+        version: &VersionRef,
+        ctx: &WriterContext<S>,
+    ) -> Option<FlushCallback> {
+        let region_id = version.metadata().id();
+        let compaction_request = CompactionRequestImpl {
+            region_id,
+            sst_layer: ctx.sst_layer.clone(),
+            writer: ctx.writer.clone(),
+            shared: ctx.shared.clone(),
+            manifest: ctx.manifest.clone(),
+            wal: ctx.wal.clone(),
+        };
+        let compaction_scheduler = ctx.compaction_scheduler.clone();
+        let shared_data = ctx.shared.clone();
+        let schedule_compaction_cb = Box::pin(async move {
+            let level0_file_num = shared_data
+                .version_control
+                .current()
+                .ssts()
+                .level(0)
+                .file_num();
+
+            // TODO(hl): Max file num in level 0 should be configurable.
+            if level0_file_num <= 10 {
```

Comment:
Could you define it as a constant?

Reply:
> Could you define it as a constant?

I added an config item `max_file_in_level0`.